test/specs/routing.test.js

```diff
@@ -402,6 +402,210 @@ module.exports = (fetchMock) => {
 					});
 				})
 			});
+
+			describe('referrer', () => {
+				it('not match when referrer not present', async () => {
+					fm
+						.mock('http://it.at.there/', 200, {
+							referrer: 'http://referrer.example.com'
+						})
+						.catch();
+
+					await fm.fetchHandler('http://it.at.there/')
+					expect(fm.calls(true).length).to.equal(0);
+				});
+
+				it('not match when referrer does\'t match', async () => {
+					fm
+						.mock('http://it.at.there/', 200, {
+							referrer: 'http://referrer.example.com'
+						})
+						.catch();
+
+					await fm.fetchHandler('http://it.at.there/', { referrer: 'http://other.referrer.example.com' })
+					expect(fm.calls(true).length).to.equal(0);
+				});
+
+				it('match when referrer option isn\'t passed as a matcher', async () => {
+					fm
+						.mock('http://it.at.there/', 200)
+						.catch();
+
+					await fm.fetchHandler('http://it.at.there/', { referrer: 'http://other.referrer.example.com' })
+					expect(fm.calls(true).length).to.equal(1);
+				});
+
+				it('match referrers', async () => {
+					fm
+						.mock('http://it.at.there/', 200, {
+							referrer: 'http://referrer.example.com'
+						})
+						.catch();
+
+					await fm.fetchHandler('http://it.at.there/', { referrer: 'http://referrer.example.com' })
+					expect(fm.calls(true).length).to.equal(1);
+				});
+
+				it('be case insensitive', async () => {
```

Comment:
should it be case insensitive? I guess probably, but worth checking if you haven't already